source/config/featureFlagEnums.py

```diff
@@ -0,0 +1,70 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+"""
+Feature flag value enumerations.
+Some feature flags require feature specific options, this file defines those options.
+All feature flags enums should
+- inherit from DisplayStringEnum and implement _displayStringLabels (for the 'displayString' property)
+- have a 'DEFAULT' member.
+"""
+import enum
+import typing
+
+from utils.displayString import (
+	DisplayStringEnum
+)
+
+from typing_extensions import Protocol  # Python 3.8 adds native support
+
+_displayString = typing.TypeVar("_displayString", bound=DisplayStringEnum)
+
+
+class FeatureFlagEnumProtocol(Protocol):
+	""" All feature flags are expected to have a "DEFAULT" value.
+	"""
+	DEFAULT: enum.Enum  # Required enum member
+
+
+FlagValueEnum = typing.Union[FeatureFlagEnumProtocol, _displayString]

```

Comment:
I believe this should be the following, based off of [this stackoverflow answer](https://stackoverflow.com/a/70907644).

```suggestion
class FeatureFlagEnumProtocol(Protocol):
	""" All feature flags are expected to have a "DEFAULT" value.
	"""
	DEFAULT: enum.Enum  # Required enum member


class FlagValueEnum(_DisplayStringEnumMixin, FeatureFlagEnumProtocol):
	pass
```

In absence of a thorough check with mypy `reveal_type`, this is what my IDE shows when using this typing structure.
![image](https://user-images.githubusercontent.com/7090342/178184052-0aee89e9-f432-4687-b2c4-308d7e85f386.png)

Reply:
Yes, it wasn't ideal for pycharm either. I haven't found an example for how to enforce this.

With the suggested approach it still isn't ideal, the type system doesn't know it is an enum (thus has `name` and `value` properties. I don't want to get held up on what might just be a limitation of the type system. For now I'll just add name and value to the `Prototype` for now.

Reply:
One issue was that the IDE didn't understand interactions with the collection, E.G. Errors with the following:
```python
	v: FlagValueEnum = get()
	theName = v.name
	theValue = v.value
	for i in v:
		print(i)
	a = v["bla"]
	isIn = a in v
```

This seems to be fixed with:
```python
class FeatureFlagEnumProtocol(Protocol, typing.Collection):
	""" All feature flags are expected to have a "DEFAULT" value.
	"""
	DEFAULT: enum.Enum  # Required enum member
	name: str  # comes from enum.Enum
	value: str  # comes from enum.Enum


class FlagValueEnum(enum.EnumMeta, _DisplayStringEnumMixin, FeatureFlagEnumProtocol):
	pass
```